pyretailscience/analysis/gain_loss.py

```diff
@@ -273,51 +274,82 @@ def plot(
         increase_cols = ["new", "increased_focus", "switch_from_comparison"]
         decrease_cols = ["lost", "decreased_focus", "switch_to_comparison"]
 
-        if self.group_col is None:
-            ax = self.gain_loss_table_df[increase_cols].plot.barh(stacked=True, color=green_colors, ax=ax, **kwargs)
-            self.gain_loss_table_df[decrease_cols].plot.barh(stacked=True, ax=ax, color=red_colors)
-            default_y_label = self.focus_group_name
+        plot_df = self.gain_loss_table_df.copy()
 
-        else:
-            ax = self.gain_loss_table_df[increase_cols].plot.barh(stacked=True, color=green_colors, **kwargs)
-            self.gain_loss_table_df[decrease_cols].plot.barh(stacked=True, ax=ax, color=red_colors)
-            default_y_label = self.group_col
+        default_y_label = self.focus_group_name if self.group_col is None else self.group_col
 
-        legend_bbox_to_anchor = None
-        if move_legend_outside:
-            legend_bbox_to_anchor = (1.05, 1)
+        plot_data = plot_df.copy()
 
-        # TODO: Ensure that each label actually has data before adding to the legend
-        legend = ax.legend(
-            [
-                "New",
-                f"Increased {self.focus_group_name}",
-                f"Switch From {self.comparison_group_name}",
-                "Lost",
-                f"Decreased {self.focus_group_name}",
-                f"Switch To {self.comparison_group_name}",
-            ],
-            frameon=True,
-            bbox_to_anchor=legend_bbox_to_anchor,
-        )
-        legend.get_frame().set_facecolor("white")
-        legend.get_frame().set_edgecolor("white")
+        color_dict = {}
+        for i, col in enumerate(increase_cols):
+            color_dict[col] = green_colors[i]
+        for i, col in enumerate(decrease_cols):
+            color_dict[col] = red_colors[i]
 
-        ax = gu.standard_graph_styles(
-            ax,
+        plot_kwargs = kwargs.copy()
+        plot_kwargs["stacked"] = True
+
+        ax = bar.plot(
+            df=plot_data,
+            value_col=increase_cols + decrease_cols,
             title=gu.not_none(title, f"Gain Loss from {self.focus_group_name} to {self.comparison_group_name}"),
             y_label=gu.not_none(y_label, default_y_label),
             x_label=gu.not_none(x_label, self.value_col),
+            orientation="horizontal",
+            ax=ax,
+            source_text=source_text,
             move_legend_outside=move_legend_outside,
+            **plot_kwargs,
```

Comment:
We've done this in a couple places and I think it's a little cleaner
```suggestion
        kwargs.pop("stacked", None)

        ax = bar.plot(
            df=plot_data,
            value_col=increase_cols + decrease_cols,
            title=gu.not_none(title, f"Gain Loss from {self.focus_group_name} to {self.comparison_group_name}"),
            y_label=gu.not_none(y_label, default_y_label),
            x_label=gu.not_none(x_label, self.value_col),
            orientation="horizontal",
            ax=ax,
            source_text=source_text,
            move_legend_outside=move_legend_outside,
            stacked=True
            **kwargs,
```